kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsFileServiceImpl.java

```diff
@@ -0,0 +1,157 @@
+package edu.arizona.kfs.vnd.service.impl;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Date;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.SuffixFileFilter;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.kuali.kfs.sys.KFSConstants;
+
+import edu.arizona.kfs.vnd.VendorConstants;
+import edu.arizona.kfs.vnd.service.EcustomsFileService;
+
+/**
+ * UAF-66 MOD-PA7000-02 ECustoms - US Export Compliance
+ *
+ * New service to handle the creation of the eCustoms data file for validating vendor compliance with US Export Controls.
+ *
+ * @author Adam Kost kosta@email.arizona.edu
+ */
+public class EcustomsFileServiceImpl implements EcustomsFileService {
+
+    private static final String ERROR_MISSING_OUTPUT_FILE_DIRECTORY = "The server property ecustoms.output.directory is blank.";
+    private static final String ERROR_MISSING_OUTPUT_FILE_PREFIX = "The server property ecustoms.output.file.prefix is blank.";
+    private static final String ERROR_CANNOT_CREATE_DIRECTORY = "Unable to create directory structure.";
+
+    private static final Logger LOG = Logger.getLogger(EcustomsFileServiceImpl.class);
+    private String outputFileDirectory;
+    private String outputFileNamePrefix;
+
+    // private getter for error checking purposes, so that this error checking can be reused whenever this field is used.
+    private String getOutputFileDirectory() throws IOException {
+        if (outputFileDirectory == null || StringUtils.isBlank(outputFileDirectory)) {

```

Comment:
For future reference, isBlank(...) also does a null check -- I realize this is a port, but we want to clean up code as we get the chance.